catalog/spatial/registry/registry-identification-plugin/src/test/java/org/codice/ddf/registry/identification/RegistryIdRetrieverTest.java

```diff
@@ -0,0 +1,133 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.identification;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import java.security.PrivilegedExceptionAction;
+import java.util.Collections;
+import java.util.concurrent.Callable;
+import org.codice.ddf.registry.common.RegistryConstants;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.security.common.Security;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RegistryIdRetrieverTest {
+
+  private RegistryIdRetriever registryIdRetriever;
+
+  private FilterBuilder builder = new GeotoolsFilterBuilder();
+
+  @Mock private CatalogFramework framework;
+
+  @Mock private Security security;
+
+  @Before
+  public void setup() {
+    registryIdRetriever = new RegistryIdRetriever(security);
+    registryIdRetriever.setCatalogFramework(framework);
+    registryIdRetriever.setFilterBuilder(builder);
+  }
+
+  @Test
+  public void testGetRegistryIdInfoNormalRegistryMetacard() throws Exception {
+    Metacard metacard = getDefaultMetacard();
+    QueryResponseImpl response =
+        new QueryResponseImpl(null, Collections.singletonList(new ResultImpl(metacard)), 1L);
+    when(security.runAsAdminWithException(any(PrivilegedExceptionAction.class)))
+        .thenAnswer(invocation -> ((PrivilegedExceptionAction) invocation.getArguments()[0]).run());
+    when(security.runWithSubjectOrElevate(any(Callable.class)))
+        .thenAnswer(invocation -> ((Callable) invocation.getArguments()[0]).call());
+    when(framework.query(any(QueryRequest.class))).thenReturn(response);
+    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
+    assertThat(idInfo.getRegistryIds().size(), equalTo(1));
+    assertThat(idInfo.getRegistryIds().iterator().next(), equalTo("regId"));
+    assertThat(idInfo.getLocalRegistryIds().size(), equalTo(0));
+    assertThat(idInfo.getRemoteMetacardIds().size(), equalTo(0));
```

Comment:
I know this wasn't you changes, but some hamcrest methods could be used for these unit tests:

```java
...
import static org.hamcrest.Matchers.contains;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.collection.IsEmptyCollection.empty;
...
  @Test
  public void testGetRegistryIdInfoNormalRegistryMetacard() throws Exception {
  ...
    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
    assertThat(idInfo.getRegistryIds(), contains("regId"));
    assertThat(idInfo.getLocalRegistryIds(), is(empty()));
    assertThat(idInfo.getRemoteMetacardIds(), is(empty()));
  }

  @Test
  public void testGetRegistryIdInfoInternalRegistryMetacard() throws Exception {
    ...
    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
    assertThat(idInfo.getRegistryIds(), contains("remoteMcardId"));
    assertThat(idInfo.getLocalRegistryIds(), is(empty()));
    assertThat(idInfo.getRemoteMetacardIds(), is(empty()));
  }

  @Test
  public void testGetRegistryIdInfoLocalRegistryMetacard() throws Exception {
  ...
    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
    assertThat(idInfo.getRegistryIds(), contains("regId"));
    assertThat(idInfo.getLocalRegistryIds(), contains("regId"));
    assertThat(idInfo.getRemoteMetacardIds(), is(empty()));
  }
...
```